.github/workflows/misc-tests.yaml

```diff
@@ -26,3 +26,54 @@ jobs:
       - name: Test sandbox configuration
         run: |
           ./tests/ci/run_presandbox_tests.sh
+
+  assert-license-statement-in-pr-description:
+    if: github.repository_owner == 'aws'
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Install jq
+      run: |
+        sudo apt-get update -o Acquire::Languages=none -o Acquire::Translation=none
+        sudo apt-get install -y jq
+
+    - name: Check PR description
+      run: |
+        # License statement we want present.
+        LICENSE_STATEMENT="By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license."
+
+        # Fetches the PR description.
+        PR_DESCRIPTION=$(curl -s -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+          https://api.github.com/repos/${{ github.repository }}/pulls/${{ github.event.pull_request.number }} | jq -r .body)
+
+        printf "PR description:\n%s" "${PR_DESCRIPTION}"
+        echo ""
+        echo ""
+        printf "Must contain:\n%s" "${LICENSE_STATEMENT}"
+        echo ""
+        echo ""
+
+        # Normalize line endings (convert CRLF to LF)
+        PR_DESCRIPTION=$(echo "${PR_DESCRIPTION}" | tr -d '\r')
+
+        # Escape quotes in PR description
+        PR_DESCRIPTION=$(echo "${PR_DESCRIPTION}" | sed 's/"/\\"/g; s/'"'"'/\\'"'"'/g')
+
+        # Remove all spaces and tabs
+        PR_DESCRIPTION=$(echo "${PR_DESCRIPTION}" | tr -d ' \t')
+        LICENSE_STATEMENT=$(echo "${LICENSE_STATEMENT}" | tr -d ' \t')
```

Comment:
Just curious, what is this needed for? The license statement is a single sentence so there should be no new lines, and a space is a space and should match right?

Reply:
I'm not sure. If I did:
1. pushing PR with correct statement in the PR description
2. modified PR statement to incorrect and then re-ran the job.
3. modified PR statement back to correct and then re-ran the job.

(1) and (2) would return as expected i.e. former succeeded, the latter failed. But (3) would not return as expected, it would fail. Removing spaces etc. seems to have removed something that got encoded unexpected(?)